indica/plasma.py

```diff
@@ -41,270 +40,139 @@ def __init__(
 
-        self.build_atomic_data()
         self.initialize_variables(n_rad, n_R, n_z)
-
-        self.equilibrium: Equilibrium
+        self.build_atomic_data()
 
     def set_equilibrium(self, equilibrium: Equilibrium):
-        """
-        Assign equilibrium object
-        """
+        """Assign equilibrium object and associated private variables"""
         self.equilibrium = equilibrium
-        self._volume.values = self.convert_in_time(
-            self.equilibrium.volume.interp(rho_poloidal=self.rho)
-        )
-        self._area.values = self.convert_in_time(
-            self.equilibrium.area.interp(rho_poloidal=self.rho)
-        )
-        self._rmjo.values = self.convert_in_time(
-            self.equilibrium.rmjo.interp(rho_poloidal=self.rho)
-        )
-        self._rmji.values = self.convert_in_time(
-            self.equilibrium.rmji.interp(rho_poloidal=self.rho)
-        )
-        self._rmag.values = self.convert_in_time(self.equilibrium.rmag)
-        self._zmag.values = self.convert_in_time(self.equilibrium.zmag)
-        self._rmin.values = (self._rmjo - self._rmji) / 2.0
 
     def set_adf11(self, adf11: dict):
         self.adf11 = adf11
 
     def initialize_variables(self, n_rad: int = 41, n_R: int = 100, n_z: int = 100):
-        """
-        Initialize all class attributes
-        """
-
-        # Machine attributes
-        R0, R1 = self.machine_dimensions[0]
-        z0, z1 = self.machine_dimensions[1]
-        self.R = format_coord(np.linspace(R0, R1, n_R), "R")
-        self.z = format_coord(np.linspace(z0, z1, n_z), "z")
-
-        index = np.arange(n_R)
-        R_midplane = np.linspace(self.R.min(), self.R.max(), n_R)
-        z_midplane = np.full_like(R_midplane, 0.0)
-        coords_midplane = {"index": index}
-        self.R_midplane = format_dataarray(R_midplane, "R_midplane", coords_midplane)
-        self.z_midplane = format_dataarray(z_midplane, "z_midplane", coords_midplane)
-
-        # Time and radial grid
-        self.rho = format_coord(np.linspace(0, 1.0, n_rad), self.rho_type)
-        self.t = format_coord(get_tlabels_dt(self.tstart, self.tend, self.dt), "t")
-        self.time_to_calculate = deepcopy(self.t)
-
-        # Elements (ions and specifics of impurities)
-        element_z, element_a, element_name, element_symbol = [], [], [], []
+        """Initialize all class attributes"""
+        # Define coordinates
+        attrs_data: dict = {}
+        R0, R1 = self.machine_conf.MACHINE_DIMS[0]
+        z0, z1 = self.machine_conf.MACHINE_DIMS[1]
+        attrs_data["R"] = np.linspace(R0, R1, n_R)
+        attrs_data["z"] = np.linspace(z0, z1, n_z)
+        attrs_data["index"] = np.arange(n_R)
+        attrs_data["R_midplane"] = np.linspace(
+            attrs_data["R"].min(), attrs_data["R"].max(), n_R
+        )
+        attrs_data["z_midplane"] = np.full_like(attrs_data["R_midplane"], 0.0)
+        attrs_data["rhop"] = np.linspace(0, 1.0, n_rad)
```

Comment:
Everything which uses rhop to interpolate is losing a dimension, since it is now a numpy array instead of a DataArray. You can see plasma.volume now has dim_0.

Reply:
Resolved in the current version.